internal/worker/objectstorefacade/worker.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package objectstorefacade
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/worker/v4"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/core/logger"
+	coreobjectstore "github.com/juju/juju/core/objectstore"
+	"github.com/juju/juju/internal/worker/fortress"
+)
+
+const (
+	// visitWaitTimeout is the maximum time to wait for a visit to the
+	// fortress to complete. This is isn't used to limit the time for the
+	// whole operation. Instead, it is used to limit the time for the visit to
+	// the fortress to complete. Anything executing during the visit is then
+	// allowed to take as long as it needs until it's complete.
+	visitWaitTimeout = 5 * time.Second
```

Comment:
Means drain has 5 seconds to complete from the time any usage of the store is attempted. Seems to me it could easily take longer...

Reply:
No, this is only for a visit. Once inside the drain, it can take as long as it wants.

Reply:
> Does the worker trying the visit bounce and just come back to try again?

Exactly this.